courses-app/src/app/shared/registration/registration.component.html

```diff
@@ -0,0 +1,52 @@
+<form class="login" #formRegistration="ngForm" [formGroup]="registrationForm" (ngSubmit)="onFormSubmit()" >
+  <h2 class="title">Registration</h2>
+  <label class="label">Name
+    <input type="text" formControlName="name" class="form-control" required/>
+    <div *ngIf="name?.invalid && (formRegistration.submitted || name?.touched)"
+         class="alert alert-danger">
+      <div *ngIf="name?.errors?.['required']" class="invalid-feedback">
+        Name is required.
+      </div>
+      <div *ngIf="name?.errors?.['minlength']" class="invalid-feedback">
+        Name must be at least 6 characters long.
+      </div>
+    </div>
+  </label>
+
+  <label  class="label">Email
+    <input type="text" formControlName="email" class="form-control" required />
+    <div *ngIf="email?.invalid && (formRegistration.submitted || email?.touched)"
+         class="alert alert-danger">
+      <div *ngIf="email?.errors?.['required']" class="invalid-feedback">
+        Email is required.
+      </div>
+      <div *ngIf="email?.errors?.['errorText']" class="invalid-feedback">
+        Email is invalid
+      </div>
+    </div>
+  </label>
+
+
+  <label class="label">Password
+    <input type="password" formControlName="password" class="form-control" required/>
+    <div *ngIf="password?.invalid && (formRegistration.submitted || password?.touched)"
+         class="alert alert-danger">
+      <div *ngIf="password?.errors?.['required']" class="invalid-feedback">
+        password is required.
+      </div>
+      <div *ngIf="password?.errors?.['errorText']" class="invalid-feedback">
+        Password is invalid
+      </div>
+    </div>
+  </label>
```

Comment:
In reactive form we do not use ngForm, because it creates "new FormGroup". In this case we create FormGroup manually in the component class. Also input data should be retrieved in the class.